### PR TITLE
Fix event leak on reuse of touchbar item

### LINKED ISSUE #12508: Encourage Touch Bar items reusability OR add a warning in the docs
If Touch Bar items are used over 10 times a `MaxListenersExceededWarning` is being emitted:
```
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 change listeners added. Use emitter.setMaxListeners() to increase limit
```

The problem: every time `BrowserWindow.setTouchBar` is called, every Touch Bar item gets an event listener for the `change` event. However this event is never detached from those items.

Example code for achieving this warning (`mainWindow` is a `BrowserWindow` instance)
```
const button = new TouchBarButton({
	label: 'Test',
	backgroundColor: '#32DB7E'
});
for(let i = 0; i < 11; i++){
	mainWindow.setTouchBar(new TouchBar([
		button
	]));
}
```

My ask is either detach the change listener of the items when the Touch Bar is removed/changed or add a warning somewhere in the docs for people to keep in mind to not reuse Touch Bar items.

Escape hatch (very bad practice): `TouchBarItem.removeAllListeners('change')`

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Fixes #12508

### CODE REVIEW

File: lib/browser/api/touch-bar.js [hunk 1]
```diff
@@ -118,6 +118,9 @@ class TouchBar extends EventEmitter {
       window.removeListener('closed', removeListeners)
       window._touchBar = null
       delete this.windowListeners[id]
+      for (const item of this.ordereredItems) {
+        item.removeListener('change', this.changeListener)
```
Comment (ckerr) [inline lib/browser/api/touch-bar.js]: I _think_ this is undoing the `item.on('change', this.changeListener)` in `registerItem()`, but the next line there is to also register each of item.child.ordereredItems iff item.child is a TouchBar. Do we need to undo that here too?

If we don't also need that, then I'm :+1: on this PR

File: lib/browser/api/touch-bar.js [hunk 2]
```diff
@@ -118,6 +118,9 @@ class TouchBar extends EventEmitter {
       window.removeListener('closed', removeListeners)
       window._touchBar = null
       delete this.windowListeners[id]
+      for (const item of this.ordereredItems) {
```
Comment (ckerr) [inline lib/browser/api/touch-bar.js]: 'ordereredItems'? not new to this PR, but... :roll_eyes: :smile:

File: lib/browser/api/touch-bar.js (hunk 1 above)
Comment (ckerr) [reply lib/browser/api/touch-bar.js]: _edit 2:_ do we also need to remove the escapeItem changeListener too?

File: lib/browser/api/touch-bar.js (hunk 2 above)
Comment (MarshallOfSound) [reply lib/browser/api/touch-bar.js]: `items` is a map of id --> item
`orderedItems` is an array giving you the left->right order of those items containing references to the objects in the `items` map

File: lib/browser/api/touch-bar.js (hunk 1 above)
Comment (MarshallOfSound) [reply lib/browser/api/touch-bar.js]: I know clean up recursively and deal with the `escapeItem` listener

File: lib/browser/api/touch-bar.js (hunk 2 above)
Comment (ckerr) [reply lib/browser/api/touch-bar.js]: Right, I get it. Just amused by the spelling `ordereredItems` :2nd_place_medal:

Comment (MarshallOfSound) [reply lib/browser/api/touch-bar.js]: Oh, wow, ok.  That's a typo I never noticed :laughing:

**Pull request merged**